src/examples/zkapps/hello_world/run.ts

```diff
@@ -1,26 +1,214 @@
-import { HelloWorld } from './hello_world';
-import { isReady, Mina, PrivateKey, Party } from '../../../../dist/server';
+import { HelloWorld, adminPrivateKey } from './hello_world';
+import { isReady, Mina, PrivateKey, Party, Field, shutdown } from 'snarkyjs';
 
+let txn2, txn3, txn4;
 // setup local ledger
 let Local = Mina.LocalBlockchain();
 Mina.setActiveInstance(Local);
 
-// a test account that pays all the fees, and puts additional funds into the zkapp
-const deployerAccount = Local.testAccounts[0].privateKey;
+// test accounts that pays all the fees, and puts additional funds into the zkapp
+const feePayer1 = Local.testAccounts[0].privateKey;
+const feePayer2 = Local.testAccounts[1].privateKey;
+const feePayer3 = Local.testAccounts[2].privateKey;
+const feePayer4 = Local.testAccounts[3].privateKey;
 
 // zkapp account
 const zkAppPrivateKey = PrivateKey.random();
 const zkAppAddress = zkAppPrivateKey.toPublicKey();
 const zkAppInstance = new HelloWorld(zkAppAddress);
 
 console.log('Deploying Hello World ....');
-let txn = await Mina.transaction(deployerAccount, () => {
-  Party.fundNewAccount(deployerAccount);
+
+let txn = await Mina.transaction(feePayer1, () => {
+  Party.fundNewAccount(feePayer1);
   zkAppInstance.deploy({ zkappKey: zkAppPrivateKey });
 });
 
 try {
   txn.send().wait();
 } catch (err: any) {
-  console.log('Deployment failed with error', err.message);
+  throw new Error(`Deployment failed with error ${err.message}`);
+}
+
+const initialState = await Mina.getAccount(
+  zkAppAddress
+).zkapp?.appState[0].toString();
+let currentState;
+
+console.log('Initial State', initialState);
+
+// update state with value that satisfies preconditions and correct admin private key
+console.log(
+  `Updating state from ${initialState} to 4 with Admin Private Key ...`
+);
+txn = await Mina.transaction(feePayer1, () => {
+  zkAppInstance.update(Field(4), adminPrivateKey);
+  zkAppInstance.sign(zkAppPrivateKey);
+});
+
+try {
+  txn.send().wait();
+
+  currentState = await Mina.getAccount(
+    zkAppAddress
+  ).zkapp?.appState[0].toString();
+
+  if (currentState !== '4') {
+    throw new Error(
+      `Current state of ${currentState} does not match 4 after calling update with 4`
+    );
+  } else {
+    console.log(`Current state succesfully updated to ${currentState}`);
+  }
+} catch (err: any) {
+  throw new Error(`
+    Updating from ${initialState} to 4 failed with error ${err.message}`);
+}
+
+const wrongAdminPrivateKey = PrivateKey.random();
+// attempt to update state with value that satisfies preconditions and incorrect admin private key
+console.log(
+  `Attempting to update state from ${currentState} to 16 with incorrect Admin Private Key ...`
+);
+txn = await Mina.transaction(feePayer1, () => {
+  zkAppInstance.update(Field(16), wrongAdminPrivateKey);
+  zkAppInstance.sign(zkAppPrivateKey);
+});
+
+try {
+  txn.send().wait();
```

Comment:
this test case should _fail_ if `txn.send().wait()` incorrectly succeeds. currently, if it succeeds nothing bad happens.

you need something like
```ts
let correctlyFails = false;
try {
  txn.send().wait();
} catch (err) {
  correctlyFails = true;
}
if (!correctlyFails) throw Error('We could update the state with the wrong admin key');
```

BONUS: Ideally, in the catch block you should check if the error is exactly the error we expect. In this case, we expect the following line to fail:
```ts
this.account.delegate.assertEquals(adminPk);
```
I think this would throw an `assert_equal` error if the admin key is wrong, so we could check if `err.message.includes("assert_equal")`, and re-throw the original error otherwise

Reply:
Or, you can save the `correctlyFails` variable if you throw two different errors in the try block, one on the failure case and one in the (incorrect) success case, and distinguish between those errors in the catch block (same comment below)